kyo-core/jvm/src/main/scala/kyo/KyoAppPlatformSpecific.scala

```diff
@@ -0,0 +1,22 @@
+package kyo
+
+import scala.collection.mutable.ListBuffer
+
+abstract class KyoAppPlatformSpecific extends KyoApp.Base[Async & Resource & Abort[Throwable]]:
+
+    private val procListBuffer = new ListBuffer[() => Unit]
+
+    final override protected def run[A: Flat](v: => A < (Async & Resource & Abort[Throwable]))(using Frame): Unit =
+        import AllowUnsafe.embrace.danger
+        procListBuffer += (() =>
+            val result = IO.Unsafe.run(Abort.run(Async.runAndBlock(timeout)(handle(v)))).eval
+            printResult(result)
+        )
+        if maybeProc.isEmpty then
```

Comment:
Won't this only be `true` once? I think we can make the `maybeProc` a generic Iterable:

`private var initCode: Iterable[() => Unit] = Nil`

Then we can just always update it. Both versions should be able to conform to this signature.

Scala JS:
`initCode = maybeProc.toList`
jvm
`initCode = procListBuffer.toList`

Reply:
I think I understand what you're trying to say now, I'll try to update this.